src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -790,7 +830,7 @@ private void OnClusterTransactionCompletion(ClusterTransactionCommand.SingleClus
                 var options = command.Options;
 
                 RachisLogIndexNotifications.NotifyListenersAbout(index, exception);
-                ServerStore.Cluster.ClusterTransactionWaiter.TrySetException(options.TaskId, exception);
+                ServerStore.Cluster.ClusterTransactionWaiter.SetException(options.TaskId, exception);
```

Comment:
This will throw, no?

Reply:
No, because we will set it only once.

Reply:
It happens when the cluster transaction raft command is being committed only when the database is shared (that was the behavior before my changes).

Reply:
Pretty sure that you cannot guarantee that.
You may have a race condition with waiting for indexing already calling this.

Reply:
Done